pkg/query/union_test.go

```diff
@@ -33,22 +33,22 @@ func TestUnionIterator(t *testing.T) {
 		union.addSubIterator(documentAccess)
 		union.addSubIterator(multiRole)
 
-		relSeq, err := ctx.Check(union, NewObjects("document", "doc1", "doc2"), NewObject("user", "alice").WithEllipses())
+		pathSeq, err := ctx.Check(union, NewObjects("document", "doc1", "doc2"), NewObject("user", "alice").WithEllipses())
 		require.NoError(err)
 
-		rels, err := CollectAll(relSeq)
+		paths, err := CollectAll(pathSeq)
 		require.NoError(err)
 
 		// Union should contain relations from both iterators for alice on doc1 and doc2
 		// DocumentAccess: alice viewer/editor/owner on doc1, alice viewer on doc2
 		// MultiRole: alice viewer/editor/owner on doc1
 		// With the new deduplication fix, we deduplicate by resource (type + id), not by full relation
 		// So we expect only one relation per resource: 1 for doc1, 1 for doc2 (2 total)
-		require.Len(rels, 2, "Union should deduplicate to one relation per resource")
+		require.Len(paths, 2, "Union should deduplicate to one relation per resource")
 
 		// Check that we have relations for both doc1 and doc2
 		resourceIDs := make(map[string]bool)
-		for _, rel := range rels {
+		for _, rel := range paths {
 			require.Equal("alice", rel.Subject.ObjectID, "All relations should be for alice")
 			require.Equal("document", rel.Resource.ObjectType, "All relations should be for documents")
 			resourceIDs[rel.Resource.ObjectID] = true
```

Comment:
```suggestion
		for _, path := range paths {
			require.Equal("alice", rel.Subject.ObjectID, "All paths should be for alice")
			require.Equal("document", rel.Resource.ObjectType, "All paths should be for documents")
			resourceIDs[rel.Resource.ObjectID] = true
```

Reply:
also, unrelated to this PR (and keep in mind i'm still wrapping my head around the overall codebase): i don't understand the logic for deduplication. it is counter-intuitive to me to have deduplication on a union, especially when you are keeping the `relation` as the first relation in the first subiterator?